roles/openshift_clock_facts/tasks/main.yaml

```diff
@@ -0,0 +1,41 @@
+---
+- fail:
+    msg: "openshift_clock_method must be set to either 'none', 'auto', 'ntpd', or 'chrony'"
+  when:  openshift_clock_method is defined and
+    openshift_clock_method not in ["none", "auto", "ntpd", "chrony"]
+
+- fail:
+    msg: "openshift_clock_method 'ntpd' is not compatible with atomic host"
+  when:  openshift_clock_method is defined and openshift_clock_method == 'ntpd' and
+    openshift.common.is_atomic
+
+- set_fact:
+    openshift_clock_auto_method: chrony
+  when: openshift_clock_method is defined and openshift_clock_method == "auto" and
+    openshift.common.is_atomic
+
+- set_fact:
+    openshift_clock_auto_method: ntpd
+  when: openshift_clock_method is defined and openshift_clock_method == "auto" and
+    not openshift.common.is_atomic
+
+- name: Set clock facts
+  openshift_facts:
+    role: "{{ item.role }}"
+    local_facts: "{{ item.local_facts }}"
+  with_items:
+  - role: clock
+    local_facts:
+      method: "{{ openshift_clock_auto_method | default(openshift_clock_method) | default(None) }}"
+      require_sync: "{{ openshift_clock_require_sync | default(None) }}"
+      timezone: "{{ openshift_clock_timezone | default(None) }}"
```

Comment:
It might be good to set defaults for these fact values in the openshift_facts module, it would allow you to avoid having to do the openshift.clock.method is defined test after this role has been run.

Also, it looks like you can use the `timedatectl` command to query if NTP is enabled, NTP is synchronized and the current configured timezone.

It also looks like timedatectl can be used to configure ntp generically regardless of whether chrony or ntpd is installed: https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/System_Administrators_Guide/chap-Configuring_the_Date_and_Time.html